server/main.py

```diff
@@ -78,8 +79,10 @@ async def upsert_file(
 async def upsert(
     request: UpsertRequest = Body(...),
 ):
+    start = time.time()
     try:
         ids = await datastore.upsert(request.documents)
+        logger.info("Inserted new data in: " + str(time.time() - start) + "s.")
```

Comment:
Instead of using string concatenation you can use string formatting
```
elapsed_time = time.time() - start

logger.info(f"Inserted new data in: {elapsed_time:.6f}s.")
```

Reply:
Done.